src/update.ts

```diff
@@ -41,6 +46,45 @@ function getHumanReadableTimeDifference(startTime: Date): string {
   return result.join(", ");
 }
 
+function getDurationMinutes(duration: string): number {
+  const unit = duration.slice(-1).toUpperCase();
+
+  try {
+    if (unit === "M") {
+      return parseInt(duration.slice(0, -1));
+    } else if (unit === "H") {
+      return parseInt(duration.slice(0, -1)) * 60;
+    } else if (unit === "D") {
+      return parseInt(duration.slice(0, -1)) * 60 * 24;
+    } else {
+      return 0;
+    }
+  } catch (error) {
+    console.error("Error parsing duration", duration, error);
+    return 0;
+  }
+}
+
+async function closeMaintenanceIssue(octokit: Octokit, owner: string, repo: string, incidentNumber: number) {
+  await octokit.issues.unlock({
+    owner,
+    repo,
+    issue_number: incidentNumber,
+  });
+  await octokit.issues.update({
+    owner,
+    repo,
+    issue_number: incidentNumber,
+    state: "closed",
+  });
+  await octokit.issues.lock({
+    owner,
+    repo,
+    issue_number: incidentNumber,
+  });
+  console.log("Closed maintenance completed event", incidentNumber);
+}
+
 export const update = async (shouldCommit = false) => {
```

Comment:
I observed that if an error is thrown at all throughout this process, we fail the GitHub Action and dont raise any alerts.

If someone misconfigures a maintenance issue, they could inadvertently disable _all_ of their uptime checks. This probably warrants a separate issue and fix, regardless of how this PR goes.